OESS-168: Remove clang warnings.

hyoklee opened the pull request:
```
 mirror_vfd.c:2188:18: warning: format string is not a string literal
      (potentially insecure) [-Wformat-security]
        HDprintf(failure_mssg);
                 ^~~~~~~~~~~~
```

hyoklee committed:
OESS-168: Remove clang warnings.

jhendersonHDF commented on test/mirror_vfd.c:
```diff
@@ -2609,6 +2610,7 @@ main(void)
     if (nerrors == 0) {
         nerrors -= test_fapl_configuration();
         nerrors -= test_xmit_encode_decode();
+        nerrors -= test_on_disk_zoo();
```
Looks like test_on_disk_zoo is already tested on line 2615 originally. Should remove this.

lrknox reviewed (changes requested):
Remove duplicated test_on_disk_zoo()

hyoklee committed:
OESS-168: Address @lrknox review.

hyoklee commented:
BTW, test fails after addressing the review on my repo 
```
Testing File creation and immediate close                             *FAILED*
HDF5-DIAG: Error detected in HDF5 (1.13.1-1) thread 0:
  #000: H5F.c line 661 in H5Fcreate(): unable to synchronously create file
    major: File accessibility
    minor: Unable to create file
  #001: H5F.c line 615 in H5F__create_api_common(): unable to create file
    major: File accessibility
    minor: Unable to open file
  #002: H5VLcallback.c line 3428 in H5VL_file_create(): file create failed
    major: Virtual Object Layer
    minor: Unable to create file
  #003: H5VLcallback.c line 3394 in H5VL__file_create(): file create failed
    major: Virtual Object Layer
    minor: Unable to create file
  #004: H5VLnative_file.c line 94 in H5VL__native_file_create(): unable to create file
    major: File accessibility
    minor: Unable to open file
  #005: H5Fint.c line 1813 in H5F_open(): unable to open file: name = 'mirror_rw/basic_create.h5', tent_flags = 13
    major: File accessibility
    minor: Unable to open file
  #006: H5FD.c line 768 in H5FD_open(): open failed
    major: Virtual File Layer
    minor: Unable to initialize object
  #007: H5FDsplitter.c line 859 in H5FD__splitter_open(): unable to open W/O file
    major: Virtual File Layer
    minor: Unable to open file
  #008: H5FD.c line 768 in H5FD_open(): open failed
    major: Virtual File Layer
    minor: Unable to initialize object
  #009: H5FDmirror.c line 1389 in H5FD__mirror_open(): can't connect to remote server
    major: Invalid arguments to routine
    minor: Bad value
*FAILED*
   at mirror_vfd.c:1377 in test_create_and_close()...
```
See https://github.com/hyoklee/hdf5/runs/4778810698?check_suite_focus=true

lrknox commented:
All checks passed in this repo.  Do you have a different test, or different code?
________________________________
From: H. Joe Lee ***@***.***>
Sent: Tuesday, January 11, 2022 11:51 AM
To: HDFGroup/hdf5 ***@***.***>
Cc: Larry Knox ***@***.***>; Comment ***@***.***>
Subject: Re: [HDFGroup/hdf5] OESS-168: Remove clang warnings. (PR #1309)


BTW, test fails after addressing the review on my repo

Testing File creation and immediate close                             *FAILED*
HDF5-DIAG: Error detected in HDF5 (1.13.1-1) thread 0:
  #000: H5F.c line 661 in H5Fcreate(): unable to synchronously create file
    major: File accessibility
    minor: Unable to create file
  #001: H5F.c line 615 in H5F__create_api_common(): unable to create file
    major: File accessibility
    minor: Unable to open file
  #002: H5VLcallback.c line 3428 in H5VL_file_create(): file create failed
    major: Virtual Object Layer
    minor: Unable to create file
  #003: H5VLcallback.c line 3394 in H5VL__file_create(): file create failed
    major: Virtual Object Layer
    minor: Unable to create file
  #004: H5VLnative_file.c line 94 in H5VL__native_file_create(): unable to create file
    major: File accessibility
    minor: Unable to open file
  #005: H5Fint.c line 1813 in H5F_open(): unable to open file: name = 'mirror_rw/basic_create.h5', tent_flags = 13
    major: File accessibility
    minor: Unable to open file
  #006: H5FD.c line 768 in H5FD_open(): open failed
    major: Virtual File Layer
    minor: Unable to initialize object
  #007: H5FDsplitter.c line 859 in H5FD__splitter_open(): unable to open W/O file
    major: Virtual File Layer
    minor: Unable to open file
  #008: H5FD.c line 768 in H5FD_open(): open failed
    major: Virtual File Layer
    minor: Unable to initialize object
  #009: H5FDmirror.c line 1389 in H5FD__mirror_open(): can't connect to remote server
    major: Invalid arguments to routine
    minor: Bad value
*FAILED*
   at mirror_vfd.c:1377 in test_create_and_close()...


See https://github.com/hyoklee/hdf5/runs/4778810698?check_suite_focus=true

—
Reply to this email directly, view it on GitHub<https://github.com/HDFGroup/hdf5/pull/1309#issuecomment-1010213922>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/ADGMWCQ3DPQTITTSUIDKHKDUVRU2VANCNFSM5KJ7D6LA>.
Triage notifications on the go with GitHub Mobile for iOS<https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675> or Android<https://play.google.com/store/apps/details?id=com.github.android&referrer=utm_campaign%3Dnotification-email%26utm_medium%3Demail%26utm_source%3Dgithub>.
You are receiving this because you commented.Message ID: ***@***.***>

hyoklee commented:
> All checks passed in this repo. Do you have a different test, or different code?

Different test:

https://github.com/hyoklee/hdf5/blob/develop/.github/workflows/linux-auto.yml

Please add the above action and see if it passes.
